src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -25980,5 +25775,837 @@ public void Add(MyCollection c)
 
             Assert.NotNull(model.GetOperation(collectionExpression));
         }
+
+        [Fact]
+        public void SingleSpread_ListToArray()
+        {
+            var source = """
+                using System.Collections.Generic;
+
+                class C
+                {
+                    static void Main()
+                    {
+                        List<int> li = [1, 2, 3];
+                        li.Report();
+                        int[] arr = [..li];
+                        arr.Report();
+                    }
+                }
```

Comment:
>} [](http://example.com/codeflow?start=16&length=1)

Consider testing cases where the element types are considered distinct in C# but not in the runtime:
- `List<nint>` and `System.IntPtr[]`
- `List<object>` and `object?[]`
- `List<dynamic>` and `object[]`

Reply:
> - `List<dynamic>` and `object[]`

It looks like the `dynamic` case in particular is not optimized. I don't think we should try to optimize this case, but it would be good to capture the current behavior.